component/src/main/java/com/celements/calendar/Calendar.java

```diff
@@ -190,17 +186,17 @@ public SpaceReference getEventSpaceRef() {
   @Override
   public List<String> getAllowedSpaces() {
     try {
-      return getCalService().getAllowedSpaces(getDocumentReference());
+      return ImmutableList.copyOf(getCalService().getAllowedSpaces(getDocumentReference()));
     } catch (XWikiException exp) {
-      LOGGER.error("Failed to get allowed spaces for '" + getDocumentReference() + "'", 
+      LOGGER.error("Failed to get allowed spaces for '" + getDocumentReference() + "'",
           exp);
     }
-    return new ArrayList<String>();
+    return Collections.emptyList();
```

Comment:
```suggestion
    return ImmutableList.of();
```

Reply:
Collections.emptyList is already immutable. Furthermore it is a singleton.

Reply:
https://guava.dev/releases/23.0/api/docs/com/google/common/collect/ImmutableList.html#of--
> Returns the empty immutable list. This list behaves and performs comparably to Collections.emptyList(), and is preferable mainly for consistency and maintainability of your code.
ok, you have convinced me. ;-)